src/nvim/api/private/helpers.c

```diff
@@ -1342,3 +1342,185 @@ const char *get_default_stl_hl(win_T *wp)
+    }
+  } else if (opts->range.type == kObjectTypeString) {
+    if (opts->range.data.string.data[0] == '%' && opts->range.data.string.size == 1) {
+      argt |= EX_RANGE | EX_DFLALL;
+      addr_type_arg = ADDR_LINES;
+    } else {
+      api_set_error(err, kErrorTypeValidation, "Invalid value for 'range'");
+      return;
+    }
+  } else if (opts->range.type == kObjectTypeInteger) {
+    argt |= EX_RANGE | EX_ZEROR;
+    def = opts->range.data.integer;
+    addr_type_arg = ADDR_LINES;
+  } else if (HAS_KEY(opts->range)) {
+    api_set_error(err, kErrorTypeValidation, "Invalid value for 'range'");
+    return;
+  }
+
+  if (opts->count.type == kObjectTypeBoolean) {
+    if (opts->count.data.boolean) {
+      argt |= EX_COUNT | EX_ZEROR | EX_RANGE;
+      addr_type_arg = ADDR_OTHER;
+      def = 0;
+    }
+  } else if (opts->count.type == kObjectTypeInteger) {
+    argt |= EX_COUNT | EX_ZEROR | EX_RANGE;
+    addr_type_arg = ADDR_OTHER;
+    def = opts->count.data.integer;
+  } else if (HAS_KEY(opts->count)) {
+    api_set_error(err, kErrorTypeValidation, "Invalid value for 'count'");
+    return;
+  }
+
+  if (opts->addr.type == kObjectTypeString) {
+    if (parse_addr_type_arg((char_u *)opts->addr.data.string.data, (int)opts->addr.data.string.size,
+                            &addr_type_arg) != OK) {
+      api_set_error(err, kErrorTypeValidation, "Invalid value for 'addr'");
+      return;
+    }
+
+    if (addr_type_arg != ADDR_LINES) {
+      argt |= EX_ZEROR;
+    }
+  } else if (HAS_KEY(opts->addr)) {
+    api_set_error(err, kErrorTypeValidation, "Invalid value for 'addr'");
+    return;
+  }
+
+  if (opts->complete.type == kObjectTypeString) {
+    if (parse_compl_arg((char_u *)opts->complete.data.string.data,
+                        (int)opts->complete.data.string.size, &compl, &argt,
+                        (char_u **)&compl_arg) != OK) {
+      api_set_error(err, kErrorTypeValidation, "Invalid value for 'complete'");
+      return;
+    }
+  }
+
+  if (api_object_to_bool(opts->bang, "bang", false, err)) {
+    if (ERROR_SET(err)) {
+      return;
+    }
+
+    argt |= EX_BANG;
+  }
+
+  if (api_object_to_bool(opts->bar, "bar", false, err)) {
+    if (ERROR_SET(err)) {
+      return;
+    }
+
+    argt |= EX_TRLBAR;
+  }
+
+  if (api_object_to_bool(opts->register_, "register", false, err)) {
+    if (ERROR_SET(err)) {
+      return;
+    }
+
+    argt |= EX_REGSTR;
+  }
+
+  bool force = api_object_to_bool(opts->force, "force", false, err);
+  if (ERROR_SET(err)) {
+    return;
+  }
+
+  LuaRef luaref = LUA_NOREF;
+  if (opts->luaref.type == kObjectTypeLuaRef) {
+    luaref = opts->luaref.data.luaref;
+    opts->luaref.data.luaref = LUA_NOREF;
+  } else if (HAS_KEY(opts->luaref)) {
+    api_set_error(err, kErrorTypeValidation, "Invalid value for 'luaref'");
+    return;
+  }
+
+  switch (command.type) {
+  case kObjectTypeString:
+    rep = command.data.string.data;
+    break;
+  case kObjectTypeNil:
```

Comment:
hmm, or perhaps ideally, `Object command` should be allowed to be a luaref, so we can avoid the rhs to be an option.. this will require some refactor to the RPC logic (which I could look into :)

Reply:
Yea that sounds good.

Reply:
This seems to allow luaref to be used as object . Unless we don't want to allow only specific objects to be luaref this should be fun . Is there any reason not to let object be luaref ?

```diff
diff --git a/src/nvim/generators/gen_api_dispatch.lua b/src/nvim/generators/gen_api_dispatch.lua
index 21f8c3855..c4a3c754f 100644
--- a/src/nvim/generators/gen_api_dispatch.lua
+++ b/src/nvim/generators/gen_api_dispatch.lua
@@ -441,8 +441,8 @@ local function process_function(fn)
     local cparam = string.format('arg%u', j)
     local param_type = real_type(param[1])
     local lc_param_type = real_type(param[1]):lower()
-    local extra = ((param_type == "Object" or param_type == "Dictionary") and "false, ") or ""
-    if param[1] == "DictionaryOf(LuaRef)" then
+    local extra = ((param_type == "Dictionary") and "false, ") or ""
+    if param_type == "Object" or param[1] == "DictionaryOf(LuaRef)" then
       extra = "true, "
     end
     local errshift = 0
```